queue/driver_mongo.go

```diff
@@ -677,24 +677,30 @@ func (d *mongoDriver) GetAllAttempts(ctx context.Context, id string) ([]amboy.Jo
 }
 
 func (d *mongoDriver) Put(ctx context.Context, j amboy.Job) error {
-	ji, err := registry.MakeJobInterchange(j, d.opts.Format)
-	if err != nil {
-		return errors.Wrap(err, "converting in-memory job to interchange job")
-	}
-
-	ji.Scopes = j.EnqueueScopes()
+	return d.PutMany(ctx, []amboy.Job{j})
+}
 
-	d.addMetadata(ji)
+func (d *mongoDriver) PutMany(ctx context.Context, jobs []amboy.Job) error {
```

Comment:
Can we add a comment on how errors are resolved into a single error? For example, if only a subset of jobs encounter errors (some due to duplicate scopes, some due to duplicate job IDs) and the rest succeed or hit other kinds of errors, which error gets returned?